demonstrations/tutorial_liesim3.py

```diff
@@ -0,0 +1,561 @@
+            # res = ∑ (res · p_j / ||p_j||^2) * p_j 
+            value = (res @ prod).trace().real
+            value = value / (prod @ prod).trace().real
+            P1[i, j, l] = value
+            dla_and_M1_vspace.add(prod)
+
+##############################################################################
+# We want to confirm that the adjoint action of :math:`\mathcal{P}` is indeed fully described by the first moments.
+#
+# For that, we reconstruct the transformed DLA elements and compare them with the decomposition.
+
+for i, h1 in enumerate(dla):
+    res = P @ h1 @ P_dagger
+    res.simplify()
+
+    reconstruct = sum([P0[i, j] * dla[j] for j in range(dim_g)])
+    reconstruct += sum([P1[i, j, l] * dla[j] @ dla[l] for j in range(dim_g) for l in range(dim_g)])
+    reconstruct.simplify()
+
+    if res != reconstruct:
+        print(f"Missmatch: {res}, {reconstruct}")
+
+##############################################################################
+# Now that we have successfully constructed a :math:`\mathcal{P}` gate,
+# let us look how entering it in a circuit transforms DLA elements (and therefore expectation value vector elements):
+# 
+# .. math::
+# 
+#     \begin{align*}
+#     (\boldsymbol{e})_\alpha & = \text{tr}\left[h_\alpha \mathcal{P} \rho \mathcal{P}^\dagger \right] = \text{tr}\left[\mathcal{P}^\dagger h_\alpha \mathcal{P} \rho \right] \\
+#     \ & = \sum_\beta \boldsymbol{P}^0_{\alpha \beta} \text{tr}\left[ h_\beta \rho \right] + \sum_{\beta_1 \beta_2} \boldsymbol{P}^1_{\alpha \beta_1 \beta_2} \text{tr}\left[ h_{\beta_1} h_{\beta_2} \rho \right] \\
+#     \ & = \sum_\beta \boldsymbol{P}^0_{\alpha \beta} \boldsymbol{E}^0_\beta + \sum_{\beta_1 \beta_2} \boldsymbol{P}^1_{\alpha \beta_1 \beta_2} \boldsymbol{E}^1_{\beta_1 \beta_2}
+#     \end{align*}
+# 
+# Here we have defined the expectation tensor
+# :math:`(\boldsymbol{E}^m)_{\beta_1 , .. , \beta_{m+1}} = \text{tr}\left[ h_{\beta_1} .. h_{\beta_{m+1}} \rho \right]` for the :math:`m`-th moment.
+# Note that :math:`\boldsymbol{e} = \boldsymbol{E}^0` is the expectation value vector for regular :math:`\mathfrak{g}`-sim.
+#
+#
+# Such a computation corresponds to the branching off from the original diagram, with an extra contribution coming from the higher moments.
+# 
+# .. figure:: /_static/demonstration_assets/liesim3/first_split.png
+#    :width: 45%
+#    :align: center
+# 
+# When inserting an arbitrary number of DLA gates :math:`U` before and :math:`V` after the :math:`\mathcal{P}` gate,
+# we obtain the following diagram.
+# 
+# .. figure:: /_static/demonstration_assets/liesim3/first_order_diagram.png
+#    :width: 45%
+#    :align: center
+# 
+# Note that in one vertical column the :math:`U` correspond to the same matrices.
+#
+#
+# Example
+# ~~~~~~~
+#
+# Let us compute an example. For that we start by computing the initial expectation vector and tensor.
+
+E_in = [None, None]
+E_in[0] = np.zeros((dim_g), dtype=float)
+E_in[1] = np.zeros((dim_g, dim_g), dtype=float)
```

Comment:
Would this make sense?
```suggestion
E0_in = np.zeros((dim_g), dtype=float)
E1_in = np.zeros((dim_g, dim_g), dtype=float)
E_in = [E0_in, E1_in]
```